helm-charts/images/gcp-filestore-backups/gcp-filestore-backups.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import json
+import subprocess
+import time
+from datetime import datetime, timedelta
+
+import jmespath
+
+
+def main(args):
+    # Get a JSON object of the filestore backups. We filter by project and region
+    # in the gcloud command.
+    filestore_backups = subprocess.check_output(
+        [
+            "gcloud",
+            "filestore",
+            "backups",
+            "list",
+            "--format=json",
+            f"--project={args.project}",
+            f"--region={args.region}",
+        ],
+        text=True,
+    )
+    filestore_backups = json.loads(filestore_backups)
+
+    # Filter returned backups by filestore and share names
+    filestore_backups = jmespath.search(
+        f"[?sourceFileShare == '{args.filestore_share_name}' && contains(sourceInstance, '{args.filestore_name}')]",
+        filestore_backups,
+    )
+
+    # Parse `createTime` to a datetime object for comparison
+    filestore_backups = [
+        {
+            k: (
+                datetime.strptime(v.split(".")[0], "%Y-%m-%dT%H:%M:%S")
+                if k == "createTime"
+                else v
+            )
+            for k, v in backup.items()
+        }
+        for backup in filestore_backups
+    ]
+
+    # Generate a list of filestore backups that are less than 24 hours old
+    recent_filestore_backups = [
+        backup
+        for backup in filestore_backups
+        if datetime.now() - backup["createTime"] < timedelta(days=1)
+    ]
+
+    # Generate a list of filestore backups that are older than our set retention period
+    old_filestore_backups = [
+        backup
+        for backup in filestore_backups
+        if datetime.now() - backup["createTime"] > timedelta(days=args.retention_days)
+    ]
+
+    if len(recent_filestore_backups) == 0:
+        print(
+            f"There have been no recent backups of the filestore for project {args.project}. Creating a backup now..."
+        )
+
+        subprocess.check_call(
+            [
+                "gcloud",
+                "filestore",
+                "backups",
+                "create",
+                f"{args.filestore_name}-{args.filestore_share_name}-backup-{datetime.now().strftime('%Y-%m-%d')}",
+                f"--file-share={args.filestore_share_name}",
+                f"--instance={args.filestore_name}",
+                f"--instance-location={args.region}-b",
```

Comment:
I need to create another input variable for `zone` here, and maybe use `--instance-zone` instead. Adding `-b` was a fix to get this working again the 2i2c shared cluster.